### PR TITLE
feat: add Uffizzi Integration

### PR DESCRIPTION
**Pull request by ShrutiC-git**
Hi, @boojack ~
Here is the PR addressing Previews with Uffizzi - [issue](https://github.com/usememos/memos/issues/646)
This is the [PoC](https://github.com/ShrutiC-git/memos/pull/1) for[ Uffizzi](https://www.uffizzi.com/) + Memos integration and this is how the [live preview](https://app.uffizzi.com/github.com/ShrutiC-git/memos/pull/1) looks like for the PR opened in my fork.

This PR adds GitHub workflows that trigger Uffizi preview environments for any PRs to this repo — a comment, like this [here](https://github.com/ShrutiC-git/memos/pull/1#issuecomment-1334845386), is also posted on the PR which will link you to the preview environment, making it easier for maintainers and contributors to easily navigate to the preview! The previews are ephemeral, so every new PR is built in a fresh environment. 

Looking forward to helping power preview environments for Memos!

### CODE REVIEW

**Comment by waveywaves** [comment]
cc @waveywaves aka me

File: docker-compose.uffizzi.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  ingress:
+    service: memos
+    port: 5230
+
+services:
+  memos:
+    image: "${MEMOS_IMAGE}"
+    volumes:
+      - memos_volume:/var/opt/memos
+
```
Comment (boojack) [inline docker-compose.uffizzi.yml:14]: We should set dev mode in PR preview to seed some demo data.
```suggestion
services:
  memos:
    image: "${MEMOS_IMAGE}"
    volumes:
      - memos_volume:/var/opt/memos
    command: ["--mode", "dev"]

```

**Comment by boojack** [comment]
@ShrutiC-git Is there a comment with PR preview link just like render? e.g. https://github.com/usememos/memos/pull/665#issuecomment-1336036166

**Comment by ShrutiC-git** [comment]
hey there, @boojack!
thanks for the prompt feedback. I have made the change you suggested — the service now starts in dev mode. 



> @ShrutiC-git Is there a comment with PR preview link just like render? e.g. [#665 (comment)](https://github.com/usememos/memos/pull/665#issuecomment-1336036166)

Yes, this is what the comment looks like - https://github.com/ShrutiC-git/memos/pull/1#issuecomment-1334845386
In the comment, you'll get the URL of the preview environment and also a link that will take you to application logs! Once this PR is merged, you'll be able to see these comments within this repo for all the PRs opened against the repo

**Comment by ShrutiC-git** [comment]
> @ShrutiC-git And I found that the preview built memos version is not latest.
> 
> image

Oh, that's interesting, @boojack, thanks for pointing it out. The service is actually built from Dockerfile in the main branch. Perhaps, that's the issue? Wondering if it's the _dev_ mode that is overriding the build and building on an older version?

**Comment by ShrutiC-git** [comment]
hey there @boojack ~
looked into this and here are some thoughts: 
- we need to build the usememos image from source because we need changes in the code reflected in the preview. 
- because we build from source and do not pull the image from the registry, the image that is built does not have that latest tag. 
- a workaround, if we want updated and tracked images, would be to build and push on every PR action, and use this image to spin up the preview. 

What would you suggest?

**Review by boojack — APPROVED** [review]

**Pull request merged**